src/tables.jl

```diff
@@ -84,15 +102,17 @@ Any other provided `kwargs` are forwarded to an internal invocation of `Arrow.wr
 
 Note that `io_or_path` may be any type that supports `Base.write(io_or_path, bytes::Vector{UInt8})`.
 """
-function write(io_or_path, table, schema::Schema; validate::Bool=true, kwargs...)
-    # This `_columns` call is unfortunately necessary; ref https://github.com/JuliaData/Arrow.jl/issues/211
-    # It is also the case that `Tables.schema(Tables.columns(table))` is more likely to return a `Tables.Schema`
-    # (rather than `nothing`) than a bare `table`, especially if `table::Vector`. We should probably fix/improve
-    # these upstream at some point.
-    columns = _columns(table)
-    validate && Legolas.validate(columns, schema)
-    assign_to_table_metadata!(columns, (LEGOLAS_SCHEMA_QUALIFIED_METADATA_KEY => schema_qualified_string(schema),))
-    write_arrow(io_or_path, columns; kwargs...)
+function write(io_or_path, table, schema::Schema; validate::Bool=true,
+               metadata=Arrow.getmetadata(table), kwargs...)
+    validate && Legolas.validate(table, schema)
+    schema_metadata = LEGOLAS_SCHEMA_QUALIFIED_METADATA_KEY => schema_qualified_string(schema)
+    if isnothing(metadata)
+        metadata = (schema_metadata,)
+    else
+        metadata = Set(collect(metadata))
```

Comment:
I think `collect` shouldn't be necessary: `Arrow.getmetadata` returns an `AbstractDict` which iterates pairs, and `Arrow.write`'s `metadata` keyword argument says the input must interate pairs. And `Set`'s docstring says it accepts an iterable to create the set.
```suggestion
        metadata = Set(metadata)
```